src/test/osd/TestMClockClientQueue.cc

```diff
@@ -0,0 +1,187 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+
+#include <iostream>
+
+#include "gtest/gtest.h"
+#include "global/global_init.h"
+#include "common/common_init.h"
+
+#include "osd/mClockClientQueue.h"
+
+
+int main(int argc, char **argv) {
+  std::vector<const char*> args(argv, argv+argc);
+  auto cct = global_init(nullptr, args, CEPH_ENTITY_TYPE_OSD,
+			 CODE_ENVIRONMENT_UTILITY,
+			 CINIT_FLAG_NO_DEFAULT_CONFIG_FILE);
+  common_init_finish(g_ceph_context);
+
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+
+
+class MClockClientQueueTest : public testing::Test {
+public:
+  mClockClientQueue q;
+
+  entity_inst_t client1;
+  entity_inst_t client2;
+  entity_inst_t client3;
+
+  MClockClientQueueTest() :
+    q(g_ceph_context),
+    client1(entity_name_t(CEPH_ENTITY_TYPE_OSD, 1), entity_addr_t()),
+    client2(entity_name_t(CEPH_ENTITY_TYPE_OSD, 2), entity_addr_t()),
+    client3(entity_name_t(CEPH_ENTITY_TYPE_CLIENT, 1), entity_addr_t())
+  {}
+
+#if 0 // more work needed here
+  Request create_client_op(epoch_t e, const entity_inst_t& owner) {
+    return Request(spg_t(), PGQueueable(OpRequestRef(), e));
+  }
+#endif
+
+  Request create_snaptrim(epoch_t e, const entity_inst_t& owner) {
+    return Request(spg_t(),
+		   PGQueueable(PGSnapTrim(e),
+			       12, 12,
+			       utime_t(), owner, e));
+  }
+
+  Request create_scrub(epoch_t e, const entity_inst_t& owner) {
+    return Request(spg_t(),
+		   PGQueueable(PGScrub(e),
+			       12, 12,
+			       utime_t(), owner, e));
+  }
+
+  Request create_recovery(epoch_t e, const entity_inst_t& owner) {
+    return Request(spg_t(),
+		   PGQueueable(PGRecovery(e, 64),
+			       12, 12,
+			       utime_t(), owner, e));
+  }
+};
+
+
+TEST_F(MClockClientQueueTest, TestSize) {
+  ASSERT_TRUE(q.empty());
+  ASSERT_EQ(q.length(), 0u);
```

Comment:
I've addressed this and am testing it. I'll push the changes here after the tests.